cmd/importer/main.go

```diff
@@ -0,0 +1,183 @@
+	"k8s.io/client-go/kubernetes/scheme"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/log/zap"
+
+	kueue "sigs.k8s.io/kueue/apis/kueue/v1beta1"
+	"sigs.k8s.io/kueue/cmd/importer/pod"
+	"sigs.k8s.io/kueue/cmd/importer/util"
+	"sigs.k8s.io/kueue/pkg/util/useragent"
+)
+
+const (
+	NamespaceFlag      = "namespace"
+	NamespaceFlagShort = "n"
+	QueueMappingFlag   = "queuemapping"
+	QueueLabelFlag     = "queuelabel"
+	QPSFlag            = "qps"
+	BurstFlag          = "burst"
+	VerbosityFlag      = "verbose"
+	VerboseFlagShort   = "v"
+	JobsFlag           = "jobs"
+	JobsFlagShort      = "j"
+)
+
+var (
+	k8sClient client.Client
+	rootCmd   = &cobra.Command{
+		Use:   "importer",
+		Short: "Import existing (running) objects into Kueue",
+		PersistentPreRunE: func(cmd *cobra.Command, _ []string) error {
+			v, _ := cmd.Flags().GetCount(VerbosityFlag)
+			level := (v + 1) * -1
+			ctrl.SetLogger(zap.New(
+				zap.UseDevMode(true),
+				zap.ConsoleEncoder(),
+				zap.Level(zapcore.Level(level)),
+			))
+
+			kubeConfig, err := ctrl.GetConfig()
+			if err != nil {
+				return err
+			}
+			if kubeConfig.UserAgent == "" {
+				kubeConfig.UserAgent = useragent.Default()
+			}
+			qps, err := cmd.Flags().GetFloat32(QPSFlag)
+			if err != nil {
+				return err
+			}
+			kubeConfig.QPS = qps
+			bust, err := cmd.Flags().GetInt(BurstFlag)
+			if err != nil {
+				return err
+			}
+			kubeConfig.Burst = bust
+
+			if err := kueue.AddToScheme(scheme.Scheme); err != nil {
+				return err
+			}
+
+			c, err := client.New(kubeConfig, client.Options{Scheme: scheme.Scheme})
+			if err != nil {
+				return err
+			}
+			k8sClient = c
+			return nil
+		},
+		RunE: func(cmd *cobra.Command, args []string) error {
+			cmd.Flags().Visit(func(f *pflag.Flag) {
+				fmt.Println(f.Name, f.Value.String())
+			})
+			err := checkCmd(cmd, args)
+			if err != nil {
+				return err
+			}
+			return nil
+		},
+	}
+)
+
+func init() {
+	rootCmd.PersistentFlags().StringSliceP(NamespaceFlag, NamespaceFlagShort, nil, "target namespaces (at least one should be provided)")
+	rootCmd.MarkPersistentFlagRequired(NamespaceFlag)
+
+	rootCmd.PersistentFlags().String(QueueLabelFlag, "", "label used to identify the target local queue")
+	rootCmd.PersistentFlags().StringToString(QueueMappingFlag, nil, "mapping from \""+QueueLabelFlag+"\" label values to local queue names")
+	rootCmd.MarkPersistentFlagRequired(QueueLabelFlag)
+
+	rootCmd.PersistentFlags().Float32(QPSFlag, 50, "client QPS, as described in https://kubernetes.io/docs/reference/config-api/apiserver-eventratelimit.v1alpha1/#eventratelimit-admission-k8s-io-v1alpha1-Limit")
+	rootCmd.PersistentFlags().Int(BurstFlag, 50, "client Burst, https://kubernetes.io/docs/reference/config-api/apiserver-eventratelimit.v1alpha1/#eventratelimit-admission-k8s-io-v1alpha1-Limit")
+	rootCmd.PersistentFlags().UintP(JobsFlag, JobsFlagShort, 8, "number of concurrent jobs")
```

Comment:
Name it concurrency or workers

Reply:
"j" is more common for this for instance `make` is using it

Reply:
But in the context of Kueue, it is confusing, as "job" has a different meaning.

kube-scheduler uses `parallelism`

https://github.com/kubernetes/kubernetes/blob/1f3c65c5f4c79a57427297d99985ab1493db7147/pkg/scheduler/apis/config/types.go#L49

Reply:
done